ports/servoshell/desktop/app_state.rs

```diff
@@ -335,6 +335,40 @@ impl RunningAppState {
             .is_some_and(|dialogs| !dialogs.is_empty())
     }
 
+    pub(crate) fn has_active_dialog_at_webview(&self, webview_id: WebViewId) -> bool {
+        let inner = self.inner();
+        inner
+            .dialogs
+            .get(&webview_id)
+            .is_some_and(|dialogs| !dialogs.is_empty())
+    }
+
+    pub(crate) fn accept_active_dialogs(&self, webview_id: WebViewId) {
+        if let Some(dialogs) = self.inner_mut().dialogs.get_mut(&webview_id) {
+            dialogs.retain_mut(|dialog| {
+                dialog.accept();
+                false
+            });
```

Comment:
```suggestion
            dialogs.drain(..).map(|dialog| dialog.accept());
```
nit: I feel like `drain()` is better here since we won't retain anything.